nssp_demo/score_forecast.R

```diff
@@ -0,0 +1,209 @@
+#' @param ... Additional arguments passed to
+#' `scoringutils::transform_forecasts`.
+#'
+#' @return A data frame with scored forecasts and relative skill metrics.
+#' @export
+score_single_run <- function(
+    scorable_data, quantile_only_data, forecast_unit, observed, predicted,
+    sample_id = ".draw", model_col = "model", ...) {
+  forecast_sample_df <- scorable_data |>
+    scoringutils::as_forecast_sample(
+      forecast_unit = forecast_unit,
+      observed = observed,
+      predicted = predicted,
+      sample_id = sample_id
+    )
+
+  quantile_only_df <- quantile_only_data |>
+    scoringutils::as_forecast_quantile(
+      forecast_unit = forecast_unit,
+      observed = observed,
+      predicted = predicted
+    )
+  quants <- unique(quantile_only_df$quantile_level)
+  quantiles_from_samples_df <- forecast_sample_df |>
+    scoringutils::as_forecast_quantile(probs = quants)
+
+  forecast_quantile_df <-
+    dplyr::bind_rows(
+      quantile_only_df,
+      quantiles_from_samples_df
+    ) |>
+    scoringutils::as_forecast_quantile()
+
+  sample_scores <- forecast_sample_df |>
+    scoringutils::transform_forecasts(...) |>
+    scoringutils::score()
+
+  quantile_scores <- forecast_quantile_df |>
+    scoringutils::transform_forecasts(...) |>
+    scoringutils::score()
+  # Add relative skill if more than one model is present
+  if (n_distinct(scorable_data[[model_col]]) != 1) {
+    sample_scores <- scoringutils::add_relative_skill(sample_scores)
+    quantile_scores <- scoringutils::add_relative_skill(quantile_scores)
+  }
+  return(list(
+    sample_scores = sample_scores,
+    quantile_scores = quantile_scores
+  ))
+}
+
+
+prep_truth_data <- function(truth_data_path) {
+  dat <- readr::read_tsv(truth_data_path,
+    show_col_types = FALSE
+  ) |>
+    filter(data_type == "test") |>
+    rename(true_value = ed_visits) |>
+    mutate(disease = case_when(
+      disease %in% c("Influenza", "COVID-19") ~ "Disease",
+      disease == "Total" ~ "Total",
+      TRUE ~ NA
+    )) |>
+    filter(!is.na(disease)) |>
+    tidyr::pivot_wider(
+      names_from = "disease",
+      values_from = "true_value"
+    ) |>
+    mutate(prop_disease_ed_visits = Disease / Total) |>
+    tidyr::pivot_longer(
+      c(Disease, Total, prop_disease_ed_visits),
+      names_to = "disease",
+      values_to = "true_value"
+    )
+
+  return(dat)
+}
+
+read_and_score_location <- function(model_run_dir,
+                                    eval_data_filename = "eval_data",
+                                    eval_data_file_ext = "tsv") {
+  message(glue::glue("Scoring {model_run_dir}..."))
+  forecast_path <- fs::path(
+    model_run_dir,
+    "forecast_samples.parquet"
+  )
+  ts_baseline_path <- fs::path(
+    model_run_dir,
+    "baseline_ts_prop_ed_visits_forecast.parquet"
+  )
+  cdc_baseline_path <- fs::path(
+    model_run_dir,
+    "baseline_cdc_prop_ed_visits_forecast.parquet"
+  )
```

Comment:
below and throughout, we have been specifying `ext` separately.

Reply:
Changed in ed2b204